node-notes-cli/command-file.js

```diff
@@ -11,109 +11,125 @@ if (process.argv.length > 5) {
 /**
  * CLI input to read, add, update and delete notes from a JSON file.
  */
-switch (command) {
-  case 'read':
-    read();
-    break;
-  case 'create':
-    create();
-    break;
-  case 'update':
-    update();
-    break;
-  case 'delete':
-    deleteEntry();
-    break;
-  default:
-    console.error('Improper command given');
-    process.exit(1);
+try {
+  switch (command) {
+    case 'read':
+      await readAndReport();
+      break;
+    case 'create':
+      await create(input1);
+      break;
+    case 'update':
+      await update(input1, input2);
+      break;
+    case 'delete':
+      await deleteEntry(input1);
+      break;
+    default:
+      console.error('Improper command given');
+      process.exit(1);
+  }
+} catch (err) {
+  console.error('Command failed to execute:', err);
+  process.exit(1);
 }
 
 /**
- * The read function is the first step for all other functions but
- * if called to only read will log the contents of the notes.
- * @returns The parsed JSON Object to be used by other functions
+ * Reads and parses a JSON file.
+ * @returns An object
```

Comment:
It's actually a bit more specific than this since you are explicitly reading `data.json`.